homeassistant/components/starlink/__init__.py

```diff
@@ -0,0 +1,36 @@
+"""The Starlink integration."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_URL, Platform
+from homeassistant.core import HomeAssistant
+
+from .const import CONFIG_POLL_RATE, DOMAIN
+from .coordinator import StarlinkUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SENSOR, Platform.BINARY_SENSOR, Platform.BUTTON]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Starlink from a config entry."""
+    coordinator = StarlinkUpdateCoordinator(
+        hass=hass,
+        url=entry.data[CONF_URL],
+        poll_rate=entry.data[CONFIG_POLL_RATE],
+        name=entry.title,
```

Comment:
Alternatively, pass the full ConfigEntry to the coordinator.

Reply:
Is that the preferred approach? I'm relatively new to Python but I figure having a class expect things like URL instead of taking the entire config would be preferred

Reply:
I think this is less relevant now, since the coordinator takes less data. Please re-open if you disagree 🙂